### ISSUE TITLE
Suggestion: Controls

### ISSUE THREAD

**Issue by nevzheng** [evanbowman/Blind-Jump#11]
move shooting to a multidirectional scheme so you can move and shoot in different directions.
Move the aim to WASD or mouse.
Have either 360 degree shooting based on mouse, our NSEW, + diagonals based on wasd

**Comment by evanbowman** [comment]
Anything beyond what I'm doing now would be an enormous investment of my time into drawing new animations, which is why the controls are the way the are. To give you an idea, this is how many keyframes there are for the player character as of right now:

https://github.com/evanbowman/blind-jump/blob/master/res/textures/gameObjects.png

Adding support for moving and shooting in different directions would require me to make many multiples of the current set of animations, I would need to rotate the top half of the player to shoot in a certain direction and have the legs go in a different direction, and still have the top and bottom halves match up correctly, which would involve drawing more animations.

**Comment by nevzheng** [comment]
Cool! Never really thought about that

**Comment by evanbowman** [comment]
Yeah and as a side note the reason most of the game's textures are in one file is that texture switches on the graphics card result in tons of cache misses, so it's faster to throw them into all one file (commonly called a texture atlas), and then draw a huge batch of stuff by selecting sub rectangles from the same texture. Like the 4 directional shooting, a lot of this stuff is the way it is for performance and complexity reasons.

**Comment by JBobda** [comment]
Maybe add a quick guide in the settings? I saw your game on YouTube and played it and I thought it was pretty good. Getting into C++ myself (moving from Java) and I think your game is something I want to strive towards.
